Docs/source/usage/workflows/extend.rst

```diff
@@ -0,0 +1,229 @@
+.. _usage-python-extend:
+
+Extend a Simulation
```

Comment:
```suggestion
Extend a Simulation with Python
```

Reply:
I thought to keep it simple for now, because there is mostly one way, but agreed - technically we can also extend WarpX from C/C++ right now (e.g., with Cling) :sweat_smile: